devices/Bmm150/Bmm150.cs

```diff
@@ -0,0 +1,318 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Diagnostics;
+using System.IO;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Magnetometer
+{
+    /// <summary>
+    /// Bmm150 class implementing a magnetometer
+    /// </summary>
+    [Interface("Bmm150 class implementing a magnetometer")]
+    public sealed class Bmm150 : IDisposable
+    {
+        /// <summary>
+        /// I2c device comm channel
+        /// </summary>
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Bmm150 device interface
+        /// </summary>
+        private Bmm150I2cBase _Bmm150Interface;
+
+        /// <summary>
+        /// Bmm150 Trim extended register data
+        /// </summary>
+        private Bmm150TrimRegisterData _trimData;
+
+        /// <summary>
+        /// Magnetometer (R-HALL) temperature compensation value, used in axis compensation calculation functions
+        /// </summary>
+        private uint _rHall;
+
+        /// <summary>
+        /// Flag to evaluate disposal of resources
+        /// </summary>
+        private bool _shouldDispose = true;
+
+        /// <summary>
+        /// Gets or sets Magnetometer calibration compensation vector
+        /// </summary>
+        public Vector3 CalibrationCompensation { get; set; } = new Vector3();
+
+        /// <summary>
+        /// Default I2C address for the Bmm150
+        /// In the official sheet (P36) states that address is 0x13, alhtough for m5stack is 0x10
+        /// more info: https://github.com/m5stack/M5_BMM150/blob/master/src/M5_BMM150_DEFS.h#L163
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x10;
```

Comment:
please add as well a SecondaryI2cAddress = 0x10;
And is the doc says the default one is 0x13, please swap both so it will be easier for a dev to understand what is it.

Reply:
solved